File: v1/src/main/java/com/api/v1/customers/domain/CustomerRepository.java
package com.api.v1.customers.domain;

import com.api.v1.persons.domain.PersonEntity;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface CustomerRepository extends ReactiveMongoRepository<CustomerEntity, UUID> {

    @Query("{ 'person': ?0 }")
    Mono<CustomerEntity> findByPerson(PersonEntity personEntity);
}

File: v2/src/main/kotlin/com/api/v2/customers/CustomerRepository.kt
package com.api.v2.customers

import com.api.v2.persons.domain.Person
import org.springframework.data.mongodb.repository.Query
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.UUID

interface CustomerRepository: CoroutineCrudRepository<Customer,UUID> {

    @Query("{ 'person': ?0 }")
    fun findByPerson(person: Person): Customer

}